### PR TITLE
cas: support setting display name

### PR DESCRIPTION
**Pull request by tharvik**
Now, the CAS server can return an attribute stating what's the desired displayname, instead of using the username directly.

btw, one nice config block for cas' attributes would be
```yaml
cas_config:
    server_url: "https://cas-server.com"
    service_url: "https://homeserver.domain.com:8448"
    attributes:
        display_name: name
        required:
            name: value
```
but that would break existing config.

### CODE REVIEW

File: synapse/rest/client/v1/login.py [hunk 1]
```diff
@@ -414,7 +415,7 @@ def handle_cas_response(self, request, cas_response_body, client_redirect_url):
                     raise LoginError(401, "Unauthorized", errcode=Codes.UNAUTHORIZED)
 
         return self._sso_auth_handler.on_successful_auth(
-            user, request, client_redirect_url
+            user, request, client_redirect_url, displayname,
```
Comment (erikjohnston) [inline synapse/rest/client/v1/login.py]:
```suggestion
            user, request, client_redirect_url, displayname
```

This is from the liniting. You can use [`black`](https://github.com/psf/black) to autoformat btw

File: synapse/rest/client/v1/login.py [hunk 2]
```diff
@@ -447,7 +448,8 @@ def parse_cas_response(self, cas_response_body):
             raise LoginError(
                 401, "Unsuccessful CAS response", errcode=Codes.UNAUTHORIZED
             )
-        return user, attributes
+        displayname = attributes.pop(self.cas_attribute_displayname, None)
```
Comment (richvdh) [inline synapse/rest/client/v1/login.py]: surely it would make more sense to do this in `handle_cas_response` ?

File: synapse/config/cas.py [hunk 3]
```diff
@@ -43,6 +45,7 @@ def generate_config_section(self, config_dir_path, server_name, **kwargs):
         #   enabled: true
         #   server_url: "https://cas-server.com"
         #   service_url: "https://homeserver.domain.com:8448"
+        #   #attribute_displayname: name
```
Comment (richvdh) [inline synapse/config/cas.py]: feels like `displayname_attribute` would be a better name?

File: changelog.d/6114.feature [hunk 4]
```diff
@@ -0,0 +1 @@
+CAS login now provides a default display name.
```
Comment (richvdh) [inline changelog.d/6114.feature]:
```suggestion
CAS login now provides a default display name for users if a `displayname_attribute` is set in the configuration file.
```

**Pull request merged**